src/glmtools.jl

```diff
@@ -20,57 +180,157 @@ end
 function mueta{T<:Real}(::CloglogLink, η::T)
     max(eps(T), exp(η) * exp(-exp(η)))
 end
+function inverselink(::CloglogLink, η)
+    expη = exp(η)
+    μ = -expm1(-expη)
+    omμ = exp(-expη)   # the complement, 1 - μ
+    μ, max(realmin(μ), expη * omμ), max(realmin(μ), μ * omμ)
+end
 
 linkfun(::IdentityLink, μ) = μ
 linkinv(::IdentityLink, η) = η
-mueta(::IdentityLink, η) = 1
+mueta(::IdentityLink, η) = one(η)
+inverselink(::IdentityLink, η) = η, one(η), oftype(η, NaN)
 
 linkfun(::InverseLink, μ) = inv(μ)
 linkinv(::InverseLink, η) = inv(η)
 mueta(::InverseLink, η) = -inv(abs2(η))
+function inverselink(::InverseLink, η)
+    μ = inv(η)
+    μ, -abs2(μ), oftype(μ, NaN)
+end
+
+linkfun(::InverseSquareLink, μ) = inv(abs2(μ))
+linkinv(::InverseSquareLink, η) = inv(sqrt(η))
```

Comment:
I have pushed a change in the method definitions but do not define `iscanonical` to be true for the `Gamma/InverseLink` and `InverseGaussian/InverseSquareLink` combinations.